rasa/nlu/test.py

```diff
@@ -398,6 +398,33 @@ def evaluate_response_selections(
     }
 
 
+def add_confused_intents_to_report(
+    report: Dict, target_intents: Iterable[Any], predicted_intents: Iterable[Any]
+) -> Dict:
+
+    from sklearn.metrics import confusion_matrix
+    from sklearn.utils.multiclass import unique_labels
+
+    cnf_matrix = confusion_matrix(target_intents, predicted_intents)
+
+    indices = np.argsort(cnf_matrix, axis=1)
+    labels = unique_labels(target_intents, predicted_intents)
+    n_candidates = min(3, len(labels))
+
+    for label in labels:
+        if report.get(label):
+            report[label]["confused_with"] = {}
+    for i, label in enumerate(labels):
+        for j in range(n_candidates):
+            label_idx = indices[i, -j]
+            _label = labels[label_idx]
+            num_hits = int(cnf_matrix[i, label_idx])
+            if _label != label and num_hits > 0:
```

Comment:
due to the first condition you might end up with just 2 candidates even if 3 would be available.

Reply:
yeah my assumption is that 'self' will always be in the top 3, and I'd like to report the two most commonly confused ones

Reply:
Ok got it, best add this to a comment to the `min(3, candidates)`, too

Reply:
added to docstring